### ISSUE TITLE
CVS import: date field

### ISSUE THREAD

**Issue by lucavallerini** [firefly-iii/firefly-iii#1512]
Latest version of Firefly III.

Is it possible to import transactions from a CVS file setting the "date" matching with one of the dates in the CVS file? Right now if I do an import it set the date always to today, it seems it ignores the column specified as "date".

**Comment by lucavallerini** [comment]
These are some data that I want to import: I want to match the second column with the "Date" field.

```
Data Operazione;Data Valuta;Entrate;Uscite;Descrizione;Descrizione Completa
05/08/2013;02/08/2013;50,00;;ACCREDITO BONIFICO;BONIFICO
13/08/2013;13/08/2013;;25,00;RICARICA PREP;Ricarica carta prepagata
21/08/2013;21/08/2013;;25,00;RICARICA PREP;Ricarica carta prepagata
28/08/2013;27/08/2013;300,00;;VERSAMENTO FRECCIA;Accredito per bollettino bancario
29/08/2013;28/08/2013;800,00;;VERSAMENTO FRECCIA;Accredito per bollettino bancario
16/09/2013;12/09/2013;;62,50;BONUS PROMOZIONE;Accredito bonus
16/09/2013;12/09/2013;12,50;;RITENUTA PROMOZIONE;Ritenuta su bonus
01/10/2013;01/10/2013;;50,00;VS.DISPOSIZIONE;BONIFICO DA VOI DISPOSTO
14/10/2013;14/10/2013;;50,00;VS.DISPOSIZIONE;BONIFICO
21/10/2013;14/10/2013;50,00;;STORNO SCRITTURA;BONIFICO
25/10/2013;25/10/2013;;50,00;VS.DISPOSIZIONE;BONIFICO DA VOI DISPOSTO
30/10/2013;25/10/2013;50,00;;STORNO SCRITTURA;BONIFICO
04/11/2013;04/11/2013;;15,00;VS.DISPOSIZIONE;BONIFICO DA VOI DISPOSTO
27/12/2013;26/12/2013;;400,00;RICARICA PREP;Ricarica carta prepagata
```

Here is my settings.
```
{
    "file-type": "csv",
    "date-format": "dmY",
    "has-headers": true,
    "delimiter": ";",
    "apply-rules": false,
    "specifics": [],
    "import-account": 2,
    "column-count": 6,
    "column-roles": [
        "date-book",
        "date-transaction",
        "amount_credit",
        "amount_debit",
        "description",
        "note"
    ],
    "column-do-mapping": [
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "column-mapping-config": []
}
```

You are right, I messed up the date format, it needs to be d/m/Y. Now the "Date" is OK, but the "Book date" is parsed wrong: e.g., the second to last transaction it is correctly imported with date 4 November 2013 but the book date is stored as 11 April 2013.

**Comment by JC5** [comment]
I noticed something similar in #1510 and I'll test your use case as well.

**Comment by JC5** [comment]
I've tested this and it should be fixed in 4.7.5.
